contracts/smart-account/modules/SessionKeyManagers/SessionKeyManagerHybrid.sol

```diff
@@ -0,0 +1,806 @@
+         */
+        assembly ("memory-safe") {
+            let offset := add(_userOpCalldata.offset, 0x4)
+            let baseOffset := offset
+
+            let dataPointer := add(baseOffset, calldataload(offset))
+            destinations.offset := add(dataPointer, 0x20)
+            destinations.length := calldataload(dataPointer)
+            offset := add(offset, 0x20)
+
+            dataPointer := add(baseOffset, calldataload(offset))
+            callValues.offset := add(dataPointer, 0x20)
+            callValues.length := calldataload(dataPointer)
+            offset := add(offset, 0x20)
+
+            dataPointer := add(baseOffset, calldataload(offset))
+            operationCalldatas.offset := add(dataPointer, 0x20)
+            operationCalldatas.length := calldataload(dataPointer)
+        }
+    }
+
+    /*********************** Session Management *******************************/
+
+    /// @inheritdoc ISessionKeyManagerModuleHybrid
+    function disableSession(bytes32 _sessionDigest) external override {
+        delete _enabledSessionsData[_sessionDigest][msg.sender];
+        emit SessionDisabled(msg.sender, _sessionDigest);
+    }
+
+    /// @inheritdoc ISessionKeyManagerModuleHybrid
+    function enableSession(SessionData calldata sessionData) external override {
+        bytes32 sessionDataDigest_ = sessionDataDigest(sessionData);
+        _enabledSessionsData[sessionDataDigest_][msg.sender] = sessionData;
+        emit SessionCreated(msg.sender, sessionDataDigest_, sessionData);
+    }
+
+    /// @inheritdoc ISessionKeyManagerModuleHybrid
+    function enabledSessionsData(
+        bytes32 _sessionDataDigest,
+        address _sa
+    ) external view override returns (SessionData memory data) {
+        data = _enabledSessionsData[_sessionDataDigest][_sa];
+    }
+
+    /********************** ISignatureValidator ****************************/
+
+    /// @inheritdoc ISignatureValidator
+    function isValidSignature(
+        bytes32,
+        bytes memory
+    ) public pure virtual override returns (bytes4) {
+        return 0xffffffff; // do not support it here
+    }
+
+    /// @inheritdoc ISignatureValidator
+    function isValidSignatureUnsafe(
+        bytes32,
+        bytes memory
+    ) public pure virtual override returns (bytes4) {
+        return 0xffffffff; // do not support it here
+    }
+
+    /***************************** Common ***********************************/
+
+    function _verifySessionEnableDataSignature(
+        bytes calldata _sessionEnableData,
+        bytes calldata _sessionEnableSignature,
+        address _smartAccount
+    ) internal view {
+        // Verify the signature on the session enable data
+        bytes32 sessionEnableDataDigest = keccak256(_sessionEnableData);
+        if (
+            ISignatureValidator(_smartAccount).isValidSignature(
+                sessionEnableDataDigest,
+                _sessionEnableSignature
+            ) != EIP1271_MAGIC_VALUE
```

Comment:
do we support 1271 for session key sigs?

Reply:
Yes the session enable signature is a ERC1271 Signature verified by the ECDSAOwnership module